src/backend/distributed/commands/table.c

```diff
@@ -796,18 +796,6 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 		return PreprocessAlterTableStmt(node, alterTableCommand, processUtilityContext);
 	}
 
-	if (AlterTableDropsForeignKey(alterTableStatement))
-	{
-		/*
-		 * The foreign key graph keeps track of the foreign keys including local tables.
-		 * So, even if a foreign key on a local table is dropped, we should invalidate
-		 * the graph so that the next commands can see the graph up-to-date.
-		 * We are aware that utility hook would still invalidate foreign key graph
-		 * even when command fails, but currently we are ok with that.
-		 */
-		MarkInvalidateForeignKeyGraph();
-	}
-
```

Comment:
I'm kind of wondering about the rest of this function. Is it always guaranteed to be called? (e.g. for drop schema?)